src/main/java/edu/hm/hafner/analysis/parser/GnuFortranParser.java

```diff
@@ -6,46 +6,125 @@
+    private static final Pattern ERROR_MESSAGE_PATTERN = Pattern.compile("(Warning|Error|Fatal Error|Internal Error at \\(1\\)):\\s?(.*)");
 
     /**
      * Creates a new instance of {@link GnuFortranParser}.
      */
     public GnuFortranParser() {
-        super(GFORTRAN_MSG_PATTERN, true);
+        super(MESSAGE_START_PATTERN);
     }
 
+    /**
+     * This Function is called once the start of a possible error message is detected.
+     * It uses the provided lookaheadStream to check if the following lines match the message too.
+     * The lines from the lookaheadStream are only consumed if they are successfully matched to a part
+     * of the error message, if they can not be matched the function returns without creating an issue.
+     *
+     * @param matcher the regular expression matcher
+     * @param lookahead the lookahead stream to read additional lines
+     * @param builder the issue builder to use
+     *
+     * @return Issue if the rest of the message was sucesfully matched too
+     */
     @Override
-    protected Optional<Issue> createIssue(final Matcher matcher, final IssueBuilder builder) {
-        String category = LINE_PATTERN.matcher(matcher.group(5)).replaceAll("");
-        return builder.setFileName(matcher.group(1))
-                .setColumnStart(matcher.group(3))
-                .setColumnEnd(matcher.group(4))
-                .setLineStart(matcher.group(2))
+    protected Optional<Issue> createIssue(final Matcher matcher, final LookaheadStream lookahead, final IssueBuilder builder) {
+        // Gather location of the error.
+        String fileName = matcher.group(1);
+        String lineStart = matcher.group(2);
+        String columStart = matcher.group(3);
+        String columEnd = matcher.group(4);
+
+        if (!lookahead.hasNext()) {
+            // This does not match the whole message, return without issue.
+            return Optional.empty();
+        }
+        String currentLine = lookahead.peekNext();
+
+        // Match any number of include lines.
+        while (INCLUDE_LINE_PATTERN.matcher(currentLine).matches()) {
+            lookahead.next(); // Only consume line if it matched.
+            if (!lookahead.hasNext()) {
+                return Optional.empty();
+            }
+            currentLine = lookahead.peekNext();
+        }
+
+        // Optional include lines are followed by one empty line.
+        if (!currentLine.isEmpty()) {
+            return Optional.empty();
+        }
+        lookahead.next(); // Consume the empty line.
+
+        // Check for two non empty lines now, one for the offending line one for a numbered indicator.
+        if (!lookahead.hasNext() || !NON_EMPTY_LINE_PATTERN.matcher(lookahead.peekNext()).matches()) {
+            return Optional.empty();
+        }
+        lookahead.next(); // Consume after match.
+
+        if (!lookahead.hasNext() || !NON_EMPTY_LINE_PATTERN.matcher(lookahead.peekNext()).matches()) {
```

Comment:
This can be simplified. Avoid to many logic inversions.

Reply:
done.